### PR TITLE
Cleaned up 2 more problems

### PR DESCRIPTION
**Pull request by SansPapyrus683**
Solves some of the problems in #806
We're getting there :)

(and yes I have tested all of the new solutions, they get AC'd)

### CODE REVIEW

**Review by bqi343 — CHANGES_REQUESTED** [review]
indentation uses a mix of spaces and tabs

File: solutions/CSES Subarray Divisibility.mdx [hunk 1]
```diff
@@ -39,156 +39,40 @@ The answer is just the sum of this quantity over all $i$.
...
-
+/**
+ * @author Qi Wang
+ * (detemplifying courtesy of Kevin Sheng)
+ */
 int main() {
 	ios_base::sync_with_stdio(0);
-	cin.tie(0); cout.tie(0);
+	cin.tie(0);
+	cout.tie(0);
```
Comment (bqi343) [inline solutions/CSES Subarray Divisibility.mdx:54]: (issue w/ original code)

`cout.tie(0)` does nothing

https://usaco.guide/general/fast-io?lang=cpp#method-2---cintie0

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/CSES Subarray Divisibility.mdx [hunk 2]
```diff
@@ -39,156 +39,40 @@ The answer is just the sum of this quantity over all $i$.
...
-	M.resize(N);
-
-	ll psums = 0;
+	vector<long long> M(N);
+	long long psums = 0;
 	M[psums] = 1;
-
-	F0R(i, N){
-		int a; re(a);
+	for (int i = 0; i < N ; i++) {
```
Comment (maggie-j-liu) [inline solutions/CSES Subarray Divisibility.mdx:62]:
```suggestion
	for (int i = 0; i < N; i++) {
```

File: solutions/CSES Subarray Divisibility.mdx [hunk 3]
```diff
@@ -39,156 +39,39 @@ The answer is just the sum of this quantity over all $i$.
...
 	}
-	ll res = 0;
-	trav(x, M){
-		//Calculating the # of pairs. This calculates the pairs without duplicates and reverse groups.
-		res += (x * (x-1))/2;
+
+	long long ans = 0;
+	for (long long x : M) {
+		// Calculating the # of pairs. This calculates the pairs without duplicates and reverse groups.
+		ans += (x * (x-1))/2;
```
Comment (maggie-j-liu) [inline solutions/CSES Subarray Divisibility.mdx:72]:
```suggestion
		ans += (x * (x - 1)) / 2;
```

File: solutions/CSES Subarray Divisibility.mdx [hunk 4]
```diff
@@ -39,156 +39,39 @@ The answer is just the sum of this quantity over all $i$.
...
-	F0R(i, N){
-		int a; re(a);
+	for (int i = 0; i < N; i++) {
+		int a;
+		cin >> a;
 		psums += a;
-		//Accounting for negative sums here
-		M[((psums%N)+N)%N]++;
+		// Remember to account for negative sums
+		M[((psums % N) + N) % N]++;
```
Comment (bqi343) [inline solutions/CSES Subarray Divisibility.mdx:66]: `(psums % N)` - parentheses not needed

File: solutions/CSES Subarray Divisibility.mdx [hunk 5]
```diff
@@ -39,156 +39,39 @@ The answer is just the sum of this quantity over all $i$.
...
 	}
-	ll res = 0;
-	trav(x, M){
-		//Calculating the # of pairs. This calculates the pairs without duplicates and reverse groups.
-		res += (x * (x-1))/2;
+
+	long long ans = 0;
+	for (long long x : M) {
+		// Calculating the # of pairs. This calculates the pairs without duplicates and reverse groups.
+		ans += (x * (x - 1)) / 2;
```
Comment (bqi343) [inline solutions/CSES Subarray Divisibility.mdx:72]: same here

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/CSES Subarray Divisibility.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/CSES Subarray Divisibility.mdx:72]: Where? If you're talking about the outer parenthesis on `x * (x - 1)`, I feel like they help make the code more explicit.

Comment (bqi343) [reply solutions/CSES Subarray Divisibility.mdx:72]: how? people should know order of operations

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/CSES Subarray Divisibility.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/CSES Subarray Divisibility.mdx:72]: Yeah, but I feel like it emphasizes the fraction part a bit more. I'll remove it then.

File: solutions/CSES Subarray Divisibility.mdx [hunk 6]
```diff
@@ -69,7 +69,7 @@ int main() {
 	long long ans = 0;
 	for (long long x : M) {
 		// Calculating the # of pairs. This calculates the pairs without duplicates and reverse groups.
-		ans += (x * (x - 1)) / 2;
+		ans += x * (x - 1) / 2;
```
Comment (jessechoe10) [inline solutions/CSES Subarray Divisibility.mdx:72]: Optional: `ans += (x*(x-1) / 2);

Comment (SansPapyrus683) [reply solutions/CSES Subarray Divisibility.mdx:72]: Don't think that's necessary.

File: solutions/CSES Subarray Sums II.mdx [hunk 7]
```diff
@@ -16,157 +16,45 @@ We can have a map that keeps track of the prefix sums. At each index $i$, we can
...
+	sums[0] = 1;
+	for (int x : T) {
+		prefix_sum += x;
+		/*
+		 * If there is a subarray with a prefix sum of prefix_sum - X,
+		 * we can exclude it from our current subarray to get the desired sum.
+		 * Thus, we can add the number of those subarrays to our answer.
+		 */
+		ans += sums[prefix_sum - X];
+		// Incrementing the count of the subarray with sum = psum
```
Comment (bqi343) [inline solutions/CSES Subarray Sums II.mdx:52]: the variable name changed, update comment accordingly

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Review by bqi343 — APPROVED** [review]

**Pull request merged**